openapi-cli/src/test/java/io/ballerina/openapi/generators/client/UtilGenerationTests.java

```diff
@@ -53,10 +53,11 @@ public void testEmptyUtilFileGen() throws IOException, BallerinaOpenApiException
         OpenAPI openAPI = codeGenerator.normalizeOpenAPI(definitionPath, true);
         BallerinaClientGenerator ballerinaClientGenerator = new BallerinaClientGenerator(openAPI, filter, false);
         SyntaxTree clientSyntaxTree = ballerinaClientGenerator.generateSyntaxTree();
-        SyntaxTree utilSyntaxTree = ballerinaClientGenerator.getBallerinaUtilGenerator().generateUtilSyntaxTree();
+        List<String> invalidFunctionNames = Arrays.asList(GET_MAP_FOR_HEADERS);
+        Assert.assertTrue(checkUtil(invalidFunctionNames,
+                ballerinaClientGenerator.getBallerinaUtilGenerator().generateUtilSyntaxTree()));
```

Comment:
In this scenario you need to check whether the `utils.bal` file only contains the `getEncodedUri` function. Nothing else. You can test by simply comparing expected and generated syntax trees